### PR TITLE
Produces correct order of columns for nested rows when user specifies a schema

### PR DESCRIPTION
**Pull request by HyukjinKwon**
#125 

This PR produces correct order of columns when user specifies a schema.

Also, this removes some extra closures that can be removed and corrects some more styles according to [scala-style-guide#pattern-matching](https://github.com/databricks/scala-style-guide#pattern-matching).

Lastly, `convertRow` is renamed to `convertObjectWithAttributes` and `convertValues` to `convertAttributes` just to be consistent with Spark (Parquet and Json data sources)

### CODE REVIEW

File: src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala [hunk 1]
```diff
@@ -211,13 +217,17 @@ private[xml] object StaxXmlParser {
...
     }
-    // The fields are sorted so `TreeMap` is used.
-    val convertedValuesMap = convertValues(valuesMap, schema)
-    val row = TreeMap((fields ++ convertedValuesMap).toSeq : _*).values.toSeq
+
+    // Here we merge both to a row.
+    val valuesMap = fieldsMap ++ attributesMap
+    valuesMap.foreach { case (f, v) =>
+      val nameToIndex = schema.map(_.name).zipWithIndex.toMap
+      nameToIndex.get(f).foreach(row.update(_, v))
```
Comment (lokm01) [inline src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala]: I think with an assignment to the row it'd be easier to spot what's happening here...

File: src/test/resources/simple-nested-objects.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+            <b>222</b>
+        </c>
+    </ROW>
+    <ROW>
+        <c>
+            <a>333</a>
+            <b>444</b>
+        </c>
+    </ROW>
+</ROWSET>
```
Comment (lokm01) [inline src/test/resources/simple-nested-objects.xml]: Missing newline?

Comment (lokm01) [inline src/test/resources/simple-nested-objects.xml]: Looks good!

File: src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala (hunk 1 above)
Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala]: I see. Thanks! I will also add a newline in the test data.

File: src/test/resources/simple-nested-objects.xml (hunk 2 above)
Comment (HyukjinKwon) [inline src/test/resources/simple-nested-objects.xml]: (Added)

**Comment by lokm01** [comment]
Looks good :+1:

**Comment by HyukjinKwon** [comment]
I am going to merge this after reviewing this multiple times by myself and @lokm01.

**Pull request closed without merging**